Release/src/http/client/http_client_asio.cpp

```diff
@@ -477,6 +718,24 @@ class asio_context : public request_context, public std::enable_shared_from_this
 
 private:
 
+    utility::string_t generate_basic_proxy_auth_header()
+    {
+        utility::string_t header;
+        
+        header.append(header_names::proxy_authorization);
+        header.append(": Basic ");
+        
+        auto credential_str = web::details::plaintext_string(new ::utility::string_t(m_http_client->client_config().proxy().credentials().username()));
+        credential_str->append(":");
+        credential_str->append(*m_http_client->client_config().proxy().credentials().decrypt());
```

Comment:
What if credentials are not set.. Should we check for cred::is_set() before accessing the credentials?